src/Uno.UI/Microsoft/UI/Xaml/Controls/SplitButton/ToggleSplitButtonAutomationPeer.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.UI.Xaml.Controls;
+using Windows.Graphics.Display;
+using Windows.UI.Xaml.Automation;
+using Windows.UI.Xaml.Automation.Peers;
+using Windows.UI.Xaml.Automation.Provider;
+
+namespace Microsoft.UI.Xaml.Automation.Peers
+{
+	public class ToggleSplitButtonAutomationPeer : FrameworkElementAutomationPeer, IExpandCollapseProvider, IToggleProvider
+	{
+		private ToggleSplitButton _owner;
+
+		public ToggleSplitButtonAutomationPeer(ToggleSplitButton owner) : base(owner)
+		{
+			_owner = owner;
+		}
+
+		// IAutomationPeerOverrides
+		protected override object GetPatternCore(PatternInterface patternInterface)
+		{
+			if (patternInterface == PatternInterface.ExpandCollapse ||
+				patternInterface == PatternInterface.Toggle)
+			{
+				return this;
+			}
+			return base.GetPatternCore(patternInterface);
+		}
+
+		protected override string GetClassNameCore()
+		{
+			return nameof(ToggleSplitButton);
+		}
+
+		protected override AutomationControlType GetAutomationControlTypeCore()
+		{
+			return AutomationControlType.SplitButton;
+		}
+
+		private ToggleSplitButton GetImpl() => _owner;
+
+		// IExpandCollapseProvider 
+		public ExpandCollapseState ExpandCollapseState
+		{
+			get
+			{
+				ExpandCollapseState currentState = ExpandCollapseState.Collapsed;
+
+				var splitButton = GetImpl();
+				if (splitButton != null)
+				{
+					if (splitButton.IsFlyoutOpen)
+					{
+						currentState = ExpandCollapseState.Expanded;
+					}
+				}
+
+				return currentState;
+			}
+		}
+
+		public void Expand() => GetImpl()?.OpenFlyout();
+
+		public void Collapse() => GetImpl()?.CloseFlyout();
+
+		// IToggleProvider
+		public ToggleState ToggleState
+		{
+			get
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add a way to break out of this property accessor's recursion.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5246693)